File: src/main/java/cn/taskeren/gtnn/machine/MTEDisassembler.java
package cn.taskeren.gtnn.machine;

import cn.taskeren.gtnn.GTNN;
import cn.taskeren.gtnn.machine.recipe.DisassemblerRecipes;
import gregtech.api.enums.SoundResource;
import gregtech.api.interfaces.tileentity.IGregTechTileEntity;
import gregtech.api.metatileentity.implementations.MTEBasicMachineWithRecipe;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;

import java.util.ArrayList;
import java.util.List;

public class MTEDisassembler extends MTEBasicMachineWithRecipe {

	public MTEDisassembler(int aID, String aName, String aNameRegional, int aTier) {
		super(
			aID,
			aName,
			aNameRegional,
			aTier,
			"Disassembles items into their components",
//			NNRecipe.Disassembler.getRecipeMap(),
			DisassemblerRecipes.DISASSEMBLER_RECIPES,
			1,
			9,
			false,
			SoundResource.NONE,
			SpecialEffects.NONE,
			"ASSEMBLER",
			new Object[]{"ACA", "WHW", "ACA", 'A', X.ROBOT_ARM, 'C', X.CIRCUIT,'W', X.WIRE, 'H', X.HULL}
		);
	}

	@Override
	public void addAdditionalTooltipInformation(ItemStack stack, List<String> tooltip) {
		tooltip.add(GTNN.NOT_DEPRECATED_TOOLTIP);
	}

	@Override
	public ArrayList<String> getSpecialDebugInfo(IGregTechTileEntity aBaseMetaTileEntity, EntityPlayer aPlayer, int aLogLevel, ArrayList<String> aList) {
		// TODO: REMOVE
		getBaseMetaTileEntity().increaseStoredEnergyUnits(10_000_000, true);
		GTNN.logger.info("Currently stored Energy: %d", getBaseMetaTileEntity().getStoredEU());
		return super.getSpecialDebugInfo(aBaseMetaTileEntity, aPlayer, aLogLevel, aList);
	}
}

File: src/main/kotlin/cn/elytra/mod/gtnn/rewind/module/disassembler/MTEDisassembler.kt
package cn.elytra.mod.gtnn.rewind.module.disassembler

import cn.elytra.mod.gtnn.GtnhNoNerf
import cn.taskeren.gtnn.machine.recipe.DisassemblerRecipes
import gregtech.api.enums.SoundResource
import gregtech.api.metatileentity.implementations.MTEBasicMachineWithRecipe
import gregtech.api.recipe.RecipeMap
import gregtech.api.recipe.RecipeMapBackend
import net.minecraft.item.ItemStack

class MTEDisassembler : MTEBasicMachineWithRecipe {

	companion object {
		val RecipeMap: RecipeMap<RecipeMapBackend> get() = DisassemblerRecipes.DISASSEMBLER_RECIPES

		val MachineCraftRecipe = arrayOf<Any>(
			"ACA",
			"WHW",
			"ACA",
			'A', X.ROBOT_ARM,
			'C', X.CIRCUIT,
			'W', X.WIRE,
			'H', X.HULL,
		)
	}

	constructor(id: Int, name: String, nameRegional: String, tier: Int) : super(
		id, name, nameRegional, tier,
		"Disassembles items into their components",
		RecipeMap, 1, 9, false,
		SoundResource.NONE, SpecialEffects.NONE, "ASSEMBLER", MachineCraftRecipe,
	)

	override fun addAdditionalTooltipInformation(stack: ItemStack, tooltip: MutableList<String>) {
		tooltip.add(GtnhNoNerf.Tooltips.ForDeprecatedMachines)
	}

}
